### ISSUE TITLE
passTLSClientCert/Infos for Forward Authentication

### ISSUE THREAD

**Issue by pdbs** [containous/traefik#4362]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

When using forward authentication, it seems that the TLS client certificate nor certificate infos (i.e. using docker & traefik.frontend.passTLSCert=true / traefik.frontend.passTLSClientCert.infos.subject.*) are forwarded to the authentication server. Would be nice, if the TCL client certificate infos could be passed to an authentication server.

**Comment by ctas582** [comment]
Hi, is anyone actively working on this issue? I've previously added this functionality on a private fork. It allows all the `passTLSClientCert` config options to be supplied to `frontends.<name>.auth.forward` and `entryPoints.<name>.auth.forward`

e.g.
```
[entryPoints.https.auth.forward]
address = "https://auth:443/"
authResponseHeaders = ["X-Remote-User", "X-Remote-Group"]

    [entryPoints.https.auth.forward.passTLSClientCert]
    pem = true

        [entryPoints.https.auth.forward.passTLSClientCert.infos]
        notBefore = true
        notAfter = true

            [entryPoints.https.auth.forward.passTLSClientCert.infos.subject]
            country = true
            domainComponent = true
            province = true
            locality = true
            organization = true
            commonName = true
            serialNumber = true
```

If you want to take a look, I've made it available at: ctas582/traefik@397af15ff13c08012ff1cd91d5c545a93de6613a

Thought I'd check if anyone else was already working on this before going to the effort of cleaning it up, adding tests/docs/submitting a PR etc. Also, not sure if you would prefer for the common cert parsing and header generation code to be refactored into a lib file, rather than remaining inside the middleware file.

**Comment by jbdoumenjou** [comment]
Hi @pdbs ,

I did some [tests](https://github.com/jbdoumenjou/passtlscert-forwardauth), I didn't have any issue with chaining the [PassTLSClientCert](https://docs.traefik.io/middlewares/passtlsclientcert/) and the [ForwardAuth](https://docs.traefik.io/v2.1/middlewares/forwardauth/) middlewares.
We can see that the header is passed to the fake authentication server.
We will close this Issue accordingly but feel free to reopen it if you think we have misunderstood your situation.
